runtime/drivers/clickhouse/clickhouse.go

```diff
@@ -310,6 +318,33 @@ func (d driver) Open(instanceID string, config map[string]any, st *storage.Clien
 		embed:      embed,
 	}
 
+	// If AWSRoleARN is provided, assume the role and cache credentials
+	if conf.AWSRoleARN != "" {
+		// Check cache first (lock, check, unlock pattern)
+		c.awsCredentialsCacheMu.Lock()
+		awsCreds := c.awsCredentials
+		c.awsCredentialsCacheMu.Unlock()
+
+		if awsCreds != nil && awsCreds.Expiration.After(time.Now().Add(5*time.Minute)) {
+			logger.Info("using cached AWS credentials for S3 access", zap.String("role_arn", conf.AWSRoleARN), zap.Time("expiration", *awsCreds.Expiration))
+		} else {
```

Comment:
This is inside the `Open` function and `c` was just instantiated in the statement above here. So `c.awsCredentials` will always be `nil` and there's no caching/refreshing happening?